src/main/java/com/appleaster/task/Task.java

```diff
@@ -0,0 +1,31 @@
+package com.appleaster.task;
+public abstract class Task {
+    protected String description;
+    protected TaskStatus status;
+    protected TaskType type;
+
+    public Task(String description, TaskType type) {
+        this.description = description;
+        this.status = TaskStatus.NOT_DONE;
+        this.type = type;
+    }
+
+    public void markAsDone() {
+        this.status = TaskStatus.DONE;
+    }
+
+    public void setCompleted(boolean completed) {
+        this.status = completed ? TaskStatus.DONE : TaskStatus.NOT_DONE;
+    }
+
+    public boolean isCompleted() {
+        return this.status == TaskStatus.DONE;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    @Override
+    public abstract String toString();
```

Comment:
As mentioned in another comment, might be good to initialise your toString() in Task.java to contain the key information that is utilised by all the following children Todo.java, Event.java, Deadline.java and Overriding them as necessary to add additional information to be more specific to the child.